### PR TITLE
Update release notes draft to version v1.23.0-alpha.4

### PR DESCRIPTION
**Pull request by AuraSinis**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR updates the Release Notes Draft to k/k v1.23.0-alpha.4

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
This is an automated PR generated from `krel The Kubernetes Release Toolbox`

### CODE REVIEW

File: releases/release-1.23/release-notes/release-notes-draft.md [hunk 1]
```diff
@@ -2,78 +2,161 @@
...
+  linking directly to the master branch. This ensures that links reference a
+  specific point in time, rather than a document that may change over time.
+  
+  See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files
+  
+  Please use the following format for linking documentation:
+  - [KEP]: <link>
+  - [Usage]: <link>
+  - [Other doc]: <link>
+  --> ([#104389](https://github.com/kubernetes/kubernetes/pull/104389), [@saschagrunert](https://github.com/saschagrunert)) [SIG Node]
```
Comment (saschagrunert) [inline releases/release-1.23/release-notes/release-notes-draft.md:21]: Okay so this seems wrong to me :thinking:

Comment (Damans227) [reply releases/release-1.23/release-notes/release-notes-draft.md:21]: Yea, something is off here. Generally, there is an individual map.yaml file generated for every release note that is updated via `Krel`. For instance, take a look at the files changed in this recent PR: https://github.com/kubernetes/sig-release/pull/1712

**Comment by Damans227** [comment]
@AuraSinis Would you be able to share the exact `Krel` command used for updating these release notes ?

**Comment by AuraSinis** [comment]
`krel release-notes --create-draft-pr --fork=AuraSinis --tag v1.23.0-alpha.4 --log-level trace`

File: releases/release-1.23/release-notes/release-notes-draft.md (hunk 1 above)
Comment (puerco) [reply releases/release-1.23/release-notes/release-notes-draft.md:21]: Even stranger is the fact that that PR has been edited:
https://github.com/kubernetes/sig-release/blob/master/releases/release-1.23/release-notes/maps/pr-104389-map.yaml

I'll take a closer look in a bit

**Comment by cici37** [comment]
/hold
I will be syncing with the author to make sure the pr is fixed before ready to review.

**Pull request closed without merging**